ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/context/KsqlSecurityContextBinderFactory.java

```diff
@@ -108,6 +104,5 @@ public KsqlSecurityContext provide() {
 
   @Override
   public void dispose(final KsqlSecurityContext ksqlSecurityContext) {
-    ksqlSecurityContext.getServiceContext().close();
```

Comment:
Oops, I didn't mean to remove it. This is the tricky line but easy to fix. The ServiceContext must be closed only when a default is not used (so it is created in the last line of the provide method).

It can be fixed by keeping a boolean flag to track if the getServiceContext() can be closed or not. Perhapsa variable like `mustClose` to False when the defaultServiceContext is used?

Reply:
This closes the admin client and doesn't make sense to do with a singleton because a subsequent request will need it.